server/interceptor/interceptor_test.go

```diff
@@ -108,13 +109,26 @@ func TestTrillianInterceptor_TreeInterception(t *testing.T) {
 			req:     &trillian.GetLatestSignedLogRootRequest{LogId: deletedTree.TreeId},
 			wantErr: true,
 		},
+		{
+			desc:      "cancelled",
+			req:       &trillian.GetTreeRequest{TreeId: logTree.TreeId},
+			cancelled: true,
+			wantErr:   true,
+		},
 	}
 
 	ctx := context.Background()
 	intercept := New(admin, quota.Noop(), false /* quotaDryRun */, nil /* mf */)
 	for _, test := range tests {
 		handler := &fakeHandler{resp: "handler response", err: test.handlerErr}
 
+		if test.cancelled {
+			// Use a context that's already been cancelled
+			newCtx, cf := context.WithTimeout(ctx, time.Second)
```

Comment:
Use `context.WithCancel()`, so you don't need to specify a timeout.

nit: call the func `cancel`.

Reply:
Renamed.

using WithCancel changes the semantics of existing tests. I don't think it should be done in this PR unless it flakes as a result of respecting deadlines now.

Reply:
It doesn't change the semantics, the end is a cancelled context anyway. Using `WithTimeout()` immediately followed by `cancel()` is the same as using `WithCancel()`, only the former has more arguments.

Reply:
I've read the code again and you're right.